tiering/memtier.c

```diff
@@ -85,8 +98,45 @@ static int destructed;
 
 static struct memtier_memory *current_memory;
 
+void *(*real_mmap)(void *, size_t, int, int, int, off_t);
+int (*real_munmap)(void *, size_t);
+static bool init = false;
+
+// only used for few initial m/callocs that calls mmap 
+static thread_local unsigned char bytes[100000];
+static thread_local size_t last_byte = 0;
+
+static thread_local void* mmap_map[1000];
+static thread_local int num_mmaps = 0;
+extern thread_local bool dont_mmap;
+
+void do_init()
+{
+    if (init) 
+        return;
+    init = true;
+
+    real_mmap = (void * (*)(void *, size_t, int, int, int, off_t))dlsym(
+        RTLD_NEXT , "mmap");
+    real_munmap = (int (*)(void *, size_t))dlsym(RTLD_NEXT , "munmap");
+    log_err("real_mmap %p", real_mmap);
+    log_err("real_munmap %p", real_munmap);
+
+    init = false;
+}
+
```

Comment:
I can see some race conditions here... What if two threads call malloc simultaneously? This gets really complicated...